tests/unittest_brain.py

```diff
@@ -484,6 +484,34 @@ def test_from_submodule_imports(self):
         inferred = next(ast_node.infer())
         self.assertIsInstance(inferred, nodes.FunctionDef)
 
+    def test_with_metaclass_subclasses_inheritance(self):
+        ast_node = builder.extract_node(
+            """
+        class A(type):
+            def test(cls):
+                return cls
+
+        class C:
+            pass
+
+        import six
+        class B(six.with_metaclass(A, C)):
+            pass
+
+        B #@
+        """
+        )
+        inferred = next(ast_node.infer())
+        self.assertIsInstance(inferred, nodes.ClassDef)
+        self.assertEqual(inferred.name, "B")
+        bases = inferred.bases
+        self.assertIsInstance(bases[0], nodes.Call)
+        ancestors = tuple(inferred.ancestors())
+        self.assertIsInstance(ancestors[0], nodes.ClassDef)
+        self.assertEqual(ancestors[0].name, "C")
+        self.assertIsInstance(ancestors[1], nodes.ClassDef)
+        self.assertEqual(ancestors[1].name, "object")
```

Comment:
Shouldn't this be `A` and not `object` here ? I don't know anything about `six.with_metaclass` :)

Reply:
@Pierre-Sassoulas first argument of six.with_metaclass is indeed the metaclass whereas the second argument is the base class. Thus i think that it is correct to not have class A in the ancestors.
What do you think about it?

Reply:
OK, it makes sense, thank you for explaining.